File: output/qwen_2.5_72b_instruct/avatar/direct_translation/python/java/atcoder_ABC152_E.java
import java.util.Scanner;
import java.math.BigInteger;

public class atcoder_ABC152_E {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int[] A = new int[N];
        for (int i = 0; i < N; i++) {
            A[i] = scanner.nextInt();
        }
        BigInteger _lcm = BigInteger.ONE;
        for (int a : A) {
            _lcm = _lcm.multiply(BigInteger.valueOf(a)).divide(_lcm.gcd(BigInteger.valueOf(a)));
        }
        BigInteger MOD = BigInteger.valueOf(1000000007);
        BigInteger sum = BigInteger.ZERO;
        for (int s : A) {
            sum = sum.add(BigInteger.valueOf(s).modInverse(MOD));
        }
        BigInteger ans = _lcm.multiply(sum).mod(MOD);
        System.out.println(ans);
    }
}

File: output/deepseek_chat/avatar/direct_translation/java/python/successful_translation/atcoder_ABC152_E.py

import math

def main():
    N = int(input())
    A = list(map(int, input().split()))
    
    lcm = 1
    for ai in A:
        lcm = lcm // math.gcd(lcm, ai) * ai
    
    mod = 1000000007
    mod_lcm = lcm % mod
    
    ans = 0
    for ai in A:
        ans = (ans + mod_div(mod_lcm, ai, mod)) % mod
    
    print(ans)

def mod_div(a, b, mod):
    return (a % mod) * mod_inv(b, mod) % mod

def mod_inv(x, mod):
    return mod_pow(x, mod - 2, mod)

def mod_pow(b, e, mod):
    ans = 1
    x = b % mod
    while e > 0:
        if e & 1:
            ans = (ans * x) % mod
        x = (x * x) % mod
        e >>= 1
    return ans

if __name__ == "__main__":
    main()

